Add 'see more' icon to Control Panel Data Editor

skfranz opened the pull request:
# Description:
Added dropdown icon to "Occurrence Trait Coding Tools" and "Import/Update Specimen Records" in Control Panel Data Editor.
Issue #2606 : Improve discoverability of options in the collection manager control panel 

# Pull Request Checklist:

# Pre-Approval

- [x] There is a description section in the pull request that details what the proposed changes do. It can be very brief if need be, but it ought to exist.
- [ ] Hotfixes should be branched off of the `master` branch and PR'd using the **merge** option (not squashed) into the `hotfix` branch.
- [x] Features and backlog bugs should be merged into the `Development` branch, **NOT** `master`
- [ ] All new text is preferably internationalized (i.e., no end-user-visible text is hard-coded on the PHP pages), and the [spreadsheet tracking internationalizations](https://docs.google.com/spreadsheets/d/133fps9w2pUCEjUA6IGCcQotk7dn9KvepMXJ2IWUZsE8/edit?usp=sharing) has been updated either with a new row or with checkmarks to existing rows.
- [ ] There are no linter errors
- [x] New features have responsive design (i.e., look aesthetically pleasing both full screen and with small or mobile screens)
- [x] [Symbiota coding standards](https://docs.google.com/document/d/1-FwCZP5Zu4f-bPwsKeVVsZErytALOJyA2szjbfSUjmc/edit?usp=sharing) have been followed
- [ ] If any files have been reformatted (e.g., by an autoformatter), the reformat is its own, separate commit in the PR
- [x] Comment which GitHub issue(s), if any does this PR address
- [ ] If this PR makes any changes that would require additional configuration of any Symbiota portals outside of the files tracked in this repository, make sure that those changes are detailed in [this document](https://docs.google.com/document/d/1T7xbXEf2bjjm-PMrlXpUBa69aTMAIROPXVqJqa2ow_I/edit?usp=sharing).

# Post-Approval

- [x] It is the code author's responsibility to merge their own pull request after it has been approved
- [x] If this PR represents a merge into the `Development` branch, remember to use the **squash & merge** option
- [ ] If this PR represents a merge into the `hotfix` branch, remember to use the **merge** option (i.e., no squash).
- [ ] If this PR represents a merge from the `Development` branch into the master branch, remember to use the **merge** option
- [ ] If this PR represents a merge from the `hotfix` branch into the `master` branch use the **squash & merge** option
  - [ ] a subsequent PR from `master` into `Development` should be made with the **merge** option (i.e., no squash).
  - [ ] **Immediately** delete the `hotfix` branch and create a new `hotfix` branch
  - [ ] increment the Symbiota version number in the symbase.php file and commit to the `hotfix` branch.
- [ ] If the dev team has agreed that this PR represents the last PR going into the `Development` branch before a tagged release (i.e., before an imminent merge into the master branch), make sure to notify the team and [lock the `Development` branch](https://github.com/BioKIC/Symbiota/settings/branches) to prevent accidental merges while QA takes place. Follow the release protocol [here](https://github.com/BioKIC/Symbiota/blob/master/docs/release-protocol.md).
- [x] Don't forget to delete your feature branch upon merge. Ignore this step as required.

Thanks for contributing and keeping it clean!

skfranz committed:
Add "see more" icon to Data Editor Control Panel

Adds 'seemore' icon to "Occurrence Trait Coding Tools" and "Import/Update Specimen Records" in collprofiles.php.

skfranz committed:
Removed <br>

Atticus29 commented on collections/misc/collprofiles.php:
```diff
@@ -389,6 +389,9 @@ function directSubmitAction(e) {
 										<a href="#" onclick="$('li.traitItem').show(); return false;">
 											<?= $LANG['TRAIT_CODING_TOOLS'] ?>
 										</a>
+										<a href="#" onclick="$('li.traitItem').show(); return false;"> 
```
We're trying to decouple the codebase from using jQuery. I just noticed that that's not in the coding standards, so I just added it. Sorry about that! I would maybe break out the onclick call to a javaScript function defined in the <script > element of collections/misc/collprofiles.php (maybe after `directSubmitAction` is defined?).

Atticus29 commented on collections/misc/collprofiles.php:
```diff
@@ -389,6 +389,9 @@ function directSubmitAction(e) {
 											<?= $LANG['TRAIT_CODING_TOOLS'] ?>
 										</a>
+										<a href="#" onclick="$('li.traitItem').show(); return false;"> 
+											<img class="seemore" src="../../images/tochild.png" style="width:13px;height:13px;">
```
We're trying to avoid inline styling when we can. There's already a <style> element in collections/misc/collprofiles.php. Since this element isn't being used on other pages yet, new classes can be added to the <style> element directly in collections/misc/collprofiles.php (e.g., the way `.quicksearch-container {` is defined).

Atticus29 commented on collections/misc/collprofiles.php:
```diff
@@ -445,6 +448,9 @@ function directSubmitAction(e) {
 									<a href="" onclick="$('li.metadataItem').show(); return false;"  >
 										<?= $LANG['OPEN_META'] ?>
 									</a>
+									<a href="#" onclick="$('li.metadataItem').show(); return false;"> 
```
See above comment.

Atticus29 commented on collections/misc/collprofiles.php:
```diff
@@ -445,6 +448,9 @@ function directSubmitAction(e) {
 										<?= $LANG['OPEN_META'] ?>
 									</a>
+									<a href="#" onclick="$('li.metadataItem').show(); return false;"> 
+										<img class="seemore" src="../../images/tochild.png" style="width:13px;height:13px;">
```
See above comment.

Atticus29 commented on collections/misc/collprofiles.php:
```diff
@@ -471,9 +477,12 @@ function directSubmitAction(e) {
 										<?= $LANG['IMPORT_SPECIMEN'] ?>
 									</a>
-									<a id="importinfo" href="https://biokic.github.io/symbiota-docs/coll_manager/upload/" title="<?php echo $LANG['MORE_INFO']; ?>" aria-label="<?php echo $LANG['MORE_INFO']; ?>">
+									<a id="importinfo" style="text-decoration:none;" href="https://biokic.github.io/symbiota-docs/coll_manager/upload/" title="<?php echo $LANG['MORE_INFO']; ?>" aria-label="<?php echo $LANG['MORE_INFO']; ?>">
```
Avoid inline styling and replace with class definitions as described above.

Atticus29 commented on collections/misc/collprofiles.php:
```diff
@@ -471,9 +477,12 @@ function directSubmitAction(e) {
 											<img src="../../images/info.png" style="width:13px;" alt="<?= $LANG['INFO_ALT'] ?>" />
-									</a><br/>
+									</a>
+									<a href="#" onclick="$('li.importItem').show(); return false;"> 
```
See above about jQuery.

Atticus29 commented on collections/misc/collprofiles.php:
```diff
@@ -471,9 +477,12 @@ function directSubmitAction(e) {
-									</a><br/>
+									</a>
+									<a href="#" onclick="$('li.importItem').show(); return false;"> 
+										<img class="seemore" src="../../images/tochild.png" style="width:13px;height:13px;">
```
See above about inline styling

Atticus29 reviewed (changes requested):
Hi, Skylar!
Works as expected! A few minor code things that should be changed, but otherwise a great start!

skfranz committed:
Added function to show more items

Added JavaScript function "showItemsList(className)" instead of jQuery function.

Added CSS classes "seemore-icon" and "link-icon."

Atticus29 commented on collections/misc/collprofiles.php:
```diff
@@ -386,11 +402,11 @@ function directSubmitAction(e) {
 									?>
 									<li>
-										<a href="#" onclick="$('li.traitItem').show(); return false;">
+										<a href="javascript:void(0)" onclick="showItemsList('traitItem')">
```
It's not super common to see javascript:void(0) in the href attribute. Apparently, this works. But I had never seen it, and it sounds like just returning false at the end of the onclick is sufficient. E.g., `href="#" onclick="showItemsList('traitItem'); return false;">`

But I'll leave it up to you whether you want to make this change or not, since it seems to be working great!

skfranz replied:
That's there since href='#' would cause the link to refresh to the top of the page when the name was clicked.

themerekat commented:
I just tested this as well! Would the user expect this to be a toggle, so you could close the list of upload types/occurrence trait coding tools?

skfranz commented:
No, not yet.

Atticus29 commented:
> I just tested this as well! Would the user expect this to be a toggle, so you could close the list of upload types/occurrence trait coding tools?

@themerekat I could make a new ticket for Skylar to do this.
